### PR TITLE
How to Sort a Python Dictionary

### PR DESCRIPTION
**Pull request by iansedano**
**Where to put new files:**

- New files should go into a top-level subfolder, named after the article slug. For example: `my-awesome-article`

**How to merge your changes:** 

1. [Make sure the CI code style tests all pass (+ run the automatic code formatter if necessary).](https://github.com/realpython/materials/blob/master/README.md)
2. Find an RP Team member on Slack and ask them to review & approve your PR.
3. Once the PR has one positive ("approved") review, GitHub lets you merge the PR.
4. 🎉

### CODE REVIEW

File: sort-python-dictionary/compare_lambda_vs_getter.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+print(
+    f"""
+{sorted_with_lambda_time=:.2f} seconds
+{sorted_with_itemgetter_time=:.2f} seconds
+itemgetter is {(
+    sorted_with_lambda_time / sorted_with_itemgetter_time
+):.2f} times faster
+""".strip()
+)
```
Comment (bzaczynski) [inline sort-python-dictionary/compare_lambda_vs_getter.py:47]: There's a small discrepancy between the materials and tutorial-drafts repositories here, but I think that we can let that slip through:

```diff
-    f"""\
+    f"""

- ):.2f} times faster"""
+ ):.2f} times faster
- ) 
+ """.strip()
+ )
```

File: sort-python-dictionary/compare_lookup_dict_vs_list.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+from timeit import timeit
+
+lookups = [15, 18, 19, 16, 6, 12, 5, 3, 9, 20, 2, 10, 13, 17, 4, 14, 11, 7, 8]
+
+list_setup = """
+from samples import list_of_dictionaries
```
Comment (bzaczynski) [inline sort-python-dictionary/compare_lookup_dict_vs_list.py:6]: In the tutorial you have this import statement at the top of the file. Does that matter? :thinking:

File: sort-python-dictionary/compare_sorting_dict_vs_list.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+from timeit import timeit
+
+list_program = "sorted(list_of_dictionaries, key=lambda item:item['age'])"
+dict_program = """
```
Comment (bzaczynski) [inline sort-python-dictionary/compare_sorting_dict_vs_list.py:4]: Your variables are named differently than in the tutorial, and there's also a difference in how you import the samples. The code formatting seems to be different, too.

File: sort-python-dictionary/nested_sort.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        return skills["python"]
+    else:
+        # Return value that is equivalent to having no Python skill
+        return 0
+
+
+print(sorted(data.items(), key=get_python, reverse=True))
+
+# Doing the same thing with lambda function and conditional expression
+sorted(
```
Comment (bzaczynski) [inline sort-python-dictionary/nested_sort.py:26]: Perhaps, we should also print this?

File: sort-python-dictionary/sorting_a_dictionary.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
+# Sorting lists with the sorted() function
```
Comment (bzaczynski) [inline sort-python-dictionary/sorting_a_dictionary.py:1]: I'm not sure about the value those code snippets bring as they're thrown in without much context.

**Review by bzaczynski — APPROVED** [review]
@iansedano I left a couple of comments, but none of them are super important. Thus, moving along :tada:

File: sort-python-dictionary/compare_lookup_dict_vs_list.py (hunk 2 above)
Comment (iansedano) [reply sort-python-dictionary/compare_lookup_dict_vs_list.py:6]: Right, this was a linter issue. Linter was saying that I had imports that I wasn't using 😕

File: sort-python-dictionary/compare_sorting_dict_vs_list.py (hunk 3 above)
Comment (iansedano) [reply sort-python-dictionary/compare_sorting_dict_vs_list.py:4]: I've made this more consistent

File: sort-python-dictionary/sorting_a_dictionary.py (hunk 5 above)
Comment (iansedano) [reply sort-python-dictionary/sorting_a_dictionary.py:1]: I've changed the comment to

> # Reviewing the sorted() function by sorting a list

**Comment by iansedano** [comment]
Thanks @bzaczynski, I've made those adjustments you requested, good eye!

**Pull request merged**